lib/galaxy/model/tags.py

```diff
@@ -90,7 +90,10 @@ def set_tags_from_list(
         item,
         new_tags_list,
         flush=True,
+        galaxy_session: Optional["GalaxySession"] = None,
     ):
+        # Override galaxy_session if passed in
+        self.galaxy_session = self.galaxy_session or galaxy_session
```

Comment:
That doesn't seem right, we should set this when instantiating the handler or pass it through completely. This seems like it would cause confusion easily.

Reply:
Thanks! That makes more sense!